058-kraft-upgrades-and-downgrades.md

```diff
@@ -0,0 +1,213 @@
+The default version corresponding to given Kafka version will be stored in `kafka-versions.yaml` in the same way we today store the default `inter.broker.protocol.version` and `log.message.format.version`.
+
+The operator will use this field to set the metadata version in the Kafka cluster.
+In the initial deployment of the Kafka cluster, it will be set using the `kafka-storage.sh` tool.
+In existing Kafka clusters, it will be queried and changed using the Kafka Admin API.
+
+The current metadata version will be tracked in the `.status` section of the Kafka CR in `.status.kafkaMetadataVersion`.
+This field will be used for validations and to track if upgrade / downgrade can be executed.
+
+##### Upgrading metadata version
+
+When upgrading the metadata version, the operator will validate the desired version and check if it is expected to work with the used Kafka version (e.g. is not higher than the Kafka version).
+If it passes the validation, Kafka Admin API will be used to upgrade the metadata.
+If the validation fails, the existing metadata version will be used and the error will be raised in log and in the `.status` section of the Kafka CR.
+
+##### Downgrading metadata version
+
+When downgrading the `metadata.version` in Kafka, the operator will always try to do a _safe_ downgrade only.
+And if the safe downgrade fails, it will report an error (in log and in `.status` section of the Kafka CR) and continue reconciliation without changing the metadata version.
+
+In case the safe downgrade is not supported, an _unsafe_ downgrade can be attempted (as explained in one of the earlier sections, the _unsafe_ downgrade is not implemented in current Kafka versions, but it is planned to be supported in the future).
+Unsafe upgrade might result in metadata loss that might cause various problems to the Kafka cluster.
+Users who decide they want to do the unsafe downgrade can do so manually:
+* Pause the reconciliation of the Kafka CR
+* Update the Kafka CR to the updated metadata version (to avoid the operator upgrading it back immediately after the reconciliation is unpaused)
+* Use the Kafka Admin API or CLI tools to do the unsafe downgrade
+* Unpause the reconciliation of the Kafka CR
+
+##### Configuring the initial metadata version
+
+When creating a new Kafka cluster, the initial metadata version can be configured using the `--release-version` option of the `kafka-storage.sh` script.
+To allow users to deploy Kafka clusters with older metadata version, Strimzi will also store the desired `metadata.version` in the per-broker configuration Config Map. 
+The value will be mounted into the Kafka pods and used when formatting the storage of a new Kafka cluster.
+That way - unlike when using environment variable - we will not need to roll the Pod every time the `metadata.version` changes (since in such cases we can update it dynamically without rolling the pods).
+
+#### Upgrades
```

Comment:
This section heading could be a bit more specific. I'm not sure if this is really much better...

```suggestion
#### Upgrading the Kafka container images
```

Reply:
I made it a bit more specific. But this chapter really describes the whole process and how it should work. Not just the rolling update for the new containers.